apps/files/src/actions/downloadAction.ts

```diff
@@ -47,7 +47,19 @@ export const action = new FileAction({
 	iconSvgInline: () => ArrowDownSvg,
 
 	enabled(nodes: Node[]) {
-		return nodes.length > 0 && nodes
+		if (nodes.length === 0) {
+			return false
+		}
+
+		// We can download direct dav files. But if we have
+		// some folders, we need to use the /apps/files/ajax/download.php
+		// endpoint, which only supports user root folder.
+		if (nodes.some(node => node.type === FileType.Folder)
+			&& !nodes.every(node => node.root?.startsWith('/files'))) {
```

Comment:
Stops at the first `false`, and is easier to read.

```suggestion
			&& nodes.some(node => !node.root?.startsWith('/files'))) {
```